scheduler-service/internal/scheduler/ratenotification.go

```diff
@@ -0,0 +1,71 @@
+package scheduler
+
+import (
+	"encoding/json"
+	"github.com/robfig/cron"
+	log "github.com/sirupsen/logrus"
+	"github.com/streadway/amqp"
+	"scheduler-service/internal/configs"
+	"scheduler-service/internal/models"
+	"scheduler-service/internal/repository"
+	"time"
+)
+
+const (
+	EventType = "RateSchedulerTriggeredEvent"
+)
+
+type RateNotificationScheduler struct {
+	config          *configs.Crons
+	channel         *amqp.Channel
+	eventRepository repository.Event
+}
+
+func NewRateNotificationScheduler(
+	config *configs.Crons,
+	channel *amqp.Channel,
+	eventRepository repository.Event,
+) *RateNotificationScheduler {
+	return &RateNotificationScheduler{
+		config:          config,
+		channel:         channel,
+		eventRepository: eventRepository,
+	}
+}
+
+func (e *RateNotificationScheduler) StartJob() {
+	c := cron.New()
+	err := c.AddFunc("*/30 * * * *", func() {
```

Comment:
I'm a bit confused cause you have cron frequency in config file, but don't use it here.
Could you please explain why?

Reply:
Also hardcoded part just for testing purpose